lib/src/base/models/models.dart

```diff
@@ -1,7 +1,17 @@
+import 'index.dart';
+
+/// Represents the various network statuses.
 enum NetworkStatus {
+  /// [initial] is passed to the [NetworkState] constructor.
   initial,
+
+  /// [loading] is passed to the [NetworkState.copyWithLoading] method.
   loading,
+
+  /// [success] is passed to the [NetworkState.copyWithSuccess] method.
   success,
+
+  /// [failure] is passed to the [NetworkState.copyWithFailure] method.
```

Comment:
Maybe just write what it's means ?
Example:
Failure indicate that something went wrong and passed to the [NetworkState.copyWithFailure] method.